### ISSUE TITLE
[Bug] while changing the quality of the video it doesn't work showing loading

### ISSUE THREAD

**Issue by Hackoak** [LecleVietnam/LeclePackageIssue#11]
if we change the video quality in the top bar it doesn't change that it is  showing only a loading on both Android & iOS, the error stack is below

This is the error on both Android & iOS

## on Android

I/flutter (10635): Data: 1280x720
I/ExoPlayerImpl(10635): Init da37347 [ExoPlayerLib/2.18.1] [generic_x86_arm, AOSP on IA Emulator, Google, 28]
E/ExoPlayerImplInternal(10635): Playback error
E/ExoPlayerImplInternal(10635):   com.google.android.exoplayer2.ExoPlaybackException: Source error
E/ExoPlayerImplInternal(10635):       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleIoException(ExoPlayerImplInternal.java:632)
E/ExoPlayerImplInternal(10635):       at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:604)
E/ExoPlayerImplInternal(10635):       at android.os.Handler.dispatchMessage(Handler.java:102)
E/ExoPlayerImplInternal(10635):       at android.os.Looper.loop(Looper.java:193)
E/ExoPlayerImplInternal(10635):       at android.os.HandlerThread.run(HandlerThread.java:65)
E/ExoPlayerImplInternal(10635):   Caused by: com.google.android.exoplayer2.upstream.FileDataSource$FileDataSourceException: java.io.FileNotFoundException: /storage/emulated/0/Android/data/com.example.sg/files/RESOLUTION=1280x720,FRAME-RATE=30.000: open failed: ENOENT (No such file or directory)
E/ExoPlayerImplInternal(10635):       at com.google.android.exoplayer2.upstream.FileDataSource.openLocalFile(FileDataSource.java:201)
E/ExoPlayerImplInternal(10635):       at com.google.android.exoplayer2.upstream.FileDataSource.open(FileDataSource.java:112)
E/ExoPlayerImplInternal(10635):       at com.google.android.exoplayer2.upstream.DefaultDataSource.open(DefaultDataSource.java:258)
E/ExoPlayerImplInternal(10635):       at com.google.android.exoplayer2.upstream.StatsDataSource.open(StatsDataSource.java:84)
E/ExoPlayerImplInternal(10635):       at com.google.android.exoplayer2.upstream.DataSourceInputStream.checkOpened(DataSourceInputStream.java:99)
E/ExoPlayerImplInternal(10635):       at com.google.android.exoplayer2.upstream.DataSourceInputStream.open(DataSourceInputStream.java:62)
E/ExoPlayerImplInternal(10635):       at com.google.android.exoplayer2.upstream.ParsingLoadable.load(ParsingLoadable.java:174)
E/ExoPlayerImplInternal(10635):       at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:412)
E/ExoPlayerImplInternal(10635):       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
E/ExoPlayerImplInternal(10635):       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
E/ExoPlayerImplInternal(10635):       at java.lang.Thread.run(Thread.java:764)
E/ExoPlayerImplInternal(10635):   Caused by: java.io.FileNotFoundException: /storage/emulated/0/Android/data/com.example.sg/files/RESOLUTION=1280x720,FRAME-RATE=30.000: open failed: ENOENT (No such file or directory)
E/ExoPlayerImplInternal(10635):       at libcore.io.IoBridge.open(IoBridge.java:485)
E/ExoPlayerImplInternal(10635):       at java.io.RandomAccessFile.<init>(RandomAccessFile.java:288)
E/ExoPlayerImplInternal(10635):       at java.io.RandomAccessFile.<init>(RandomAccessFile.java:151)
E/ExoPlayerImplInternal(10635):       at com.google.android.exoplayer2.upstream.FileDataSource.openLocalFile(FileDataSource.java:182)
E/ExoPlayerImplInternal(10635):       ... 10 more
E/ExoPlayerImplInternal(10635):   Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
E/ExoPlayerImplInternal(10635):       at libcore.io.Linux.open(Native Method)
E/ExoPlayerImplInternal(10635):       at libcore.io.BlockGuardOs.open(BlockGuardOs.java:210)
E/ExoPlayerImplInternal(10635):       at libcore.io.IoBridge.open(IoBridge.java:471)
E/ExoPlayerImplInternal(10635):       ... 13 more
I/flutter (10635): Init local file error PlatformException(VideoError, Video player had error com.google.android.exoplayer2.ExoPlaybackException: Source error, null, null)

## on IOS:

flutter: --- Quality select ---
quality : 640x360
link : https://sfux-ext.sfux.info/hls/chapter/105/1588724110/1588724110_500.m3u8
flutter: Start reading file
flutter: Video file data: #EXTM3U
#EXT-X-INDEPENDENT-SEGMENTS
#EXT-X-STREAM-INF:CLOSED-CAPTIONS=NONE,BANDWIDTH=1469712,
                  RESOLUTION=640x360,FRAME-RATE=30.000
https://sfux-ext.sfux.info/hls/chapter/105/1588724110/1588724110_500.m3u8
flutter: Data: 640x360
flutter: Init local file error PlatformException(VideoError, Failed to load video: The operation couldn’t be completed. (CoreMediaErrorDomain error -12865.), null, null)

## Refs:

![This the refs for this issue](https://user-images.githubusercontent.com/98728718/217809531-f185e642-049a-4ddd-a0c7-a0d6a0e930ba.mov)

Thank You!

**Comment by LecleVietnam** [comment]
Hi @Hackoak,
Thank you for your feedback, our team will consider this issue and fix it asap.

**Comment by LecleVietnam** [comment]
Hi @Hackoak,
We have fixed this issue and publish the fix to pub.dev. Could you please update the package's version and check this issue again? Thank you.

**Comment by Hackoak** [comment]
Thank you @LecleVietnam for solving this issue asap 🥇. It's fixed :), but sometimes it will take a long time to load. not a big deal but if it's improved it will be fine. once again Thank You!

**Comment by LecleVietnam** [comment]
Hi @Hackoak,
Thank you for your feedback, we have noticed that there is a performance issue when changing the video's resolution but we don't have enough time to fix it now. However, we will try to find a solution to improve the performance for the package in the next update asap.

By the way, we will close this issue as it has been fixed. If you have any other issues feel free to create issues here, we will try to fix them asap. Thank you.
